tests/rados/test_osd_superblock.py

```diff
@@ -0,0 +1,143 @@
+        log.info(f"List of active OSDs on the cluster: {osd_list}")
+        osd_id = random.choice(osd_list)
+        log.info(f"Random OSD chosen for the test: {osd_id}")
+
+        # Check for osd_superblock object in PG meta
+        out = objectstore_obj.list_objects(
+            osd_id=osd_id, pgid="meta", obj_name="osd_superblock"
+        )
+        log.info(f"Output of object list from PG 'meta': \n {out}")
+        assert "osd_superblock" in out
+
+        # list omap entries for the osd_superblock object
+        omap_entries = objectstore_obj.list_omap(
+            osd_id=osd_id, pgid="meta", obj="osd_superblock"
+        )
+        log.info(f"OMAP entries in object osd_superblock from PG 'meta': \n {out}")
+        assert "osd_superblock" in omap_entries
+
+        # print the data present in osd_superblock omap
+        out = objectstore_obj.get_omap(
+            osd_id=osd_id, pgid="meta", obj="osd_superblock", key="osd_superblock"
+        )
+        log.info(f"osd_superblock OMAP data: \n {out}")
+
+        # create a garbage file to corrupt osd_superblock object
+        log.info("Creating garbage data to push into osd_superblock object")
+        _data = "This is a random string to corrupt OSD superblock"
+        edit_cmd = f"echo '{_data}' > /tmp/garbage_data"
+        osd_host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+        out, _ = osd_host.exec_command(sudo=True, cmd=edit_cmd)
+
+        # preserve the original content of osd_superblock object
+        log.info("Preserving the original content of osd_superblock object")
+        objectstore_obj.get_bytes(
+            osd_id=osd_id,
+            pgid="meta",
+            obj="osd_superblock",
+            out_file="/tmp/org_osd_superblock",
+        )
+
+        # overwrite osd_superblock data
+        log.info("Corrupting osd_superblock data with garbage value")
+        objectstore_obj.set_bytes(
+            osd_id=osd_id,
+            pgid="meta",
+            obj="osd_superblock",
+            in_file="/tmp/garbage_data",
+        )
+
+        # running any COT command beyond this point should result in a crash
+        try:
+            objectstore_obj.list_objects(osd_id=osd_id, pgid="meta")
+            log.error(
+                "An Exception should have been hit during the execution of list object"
+                "COT command as OSD superblock has been corrupted"
+            )
+            raise Exception(
+                "Did not encounter excepted crash in COT tool after OSD superblock corruption"
+            )
+        except Exception as er:
+            log.error(
+                f"Failure is expected because OSD superblock has been corrupted,"
+                f"Error: \n {er}"
+            )
+
+        # restart the OSD service
+        if not rados_obj.change_osd_state(action="restart", target=osd_id):
+            log.error(f"Could not restart osd - {osd_id} successfully")
```

Comment:
Do we not want to exit the test if OSD not restarted?

Reply:
Thank you for catching this, I have added a raise Exception line